### ISSUE TITLE
Popular items ItemSelector throws exception

### ISSUE THREAD

**Issue by Bionic1251** [lenskit/lenskit#852]
evaluator.addMetric(new NDCGTopNMetric("all", "5", 5, ItemSelectors.mostPopular(100), ItemSelectors.trainingItems()));

Throws exception:
```
org.grouplens.lenskit.eval.TaskExecutionException: error in evaluation job task
	at org.grouplens.lenskit.eval.traintest.TrainTestEvalTask.runEvaluations(TrainTestEvalTask.java:532)
	at org.grouplens.lenskit.eval.traintest.TrainTestEvalTask.perform(TrainTestEvalTask.java:448)
	at org.grouplens.lenskit.eval.traintest.SimpleEvaluator.call(SimpleEvaluator.java:344)
	at Runner.main(Runner.java:133)
Caused by: com.google.common.util.concurrent.UncheckedExecutionException: com.google.common.util.concurrent.UncheckedExecutionException: java.lang.NullPointerException
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2203)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3937)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3941)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4824)
	at org.grouplens.lenskit.eval.traintest.LenskitTestUser.getRecommendations(LenskitTestUser.java:97)
	at org.grouplens.lenskit.eval.metrics.topn.NDCGTopNMetric.doMeasureUser(NDCGTopNMetric.java:113)
	at org.grouplens.lenskit.eval.metrics.topn.NDCGTopNMetric.doMeasureUser(NDCGTopNMetric.java:45)
	at org.grouplens.lenskit.eval.metrics.AbstractMetric.measureUser(AbstractMetric.java:84)
	at org.grouplens.lenskit.eval.traintest.TrainTestJob$MetricWithAccumulator.measureUser(TrainTestJob.java:245)
	at org.grouplens.lenskit.eval.traintest.TrainTestJob.runEvaluation(TrainTestJob.java:142)
	at org.grouplens.lenskit.eval.traintest.TrainTestJob.call(TrainTestJob.java:86)
	at org.grouplens.lenskit.eval.traintest.JobGraph$JobNode.call(JobGraph.java:116)
	at org.grouplens.lenskit.eval.traintest.JobGraph$JobNode.call(JobGraph.java:102)
	at org.grouplens.lenskit.util.parallel.SequentialTaskGraphExecutor.execute(SequentialTaskGraphExecutor.java:37)
	at org.grouplens.lenskit.eval.traintest.TrainTestEvalTask.runEvaluations(TrainTestEvalTask.java:529)
	... 3 more
Caused by: com.google.common.util.concurrent.UncheckedExecutionException: java.lang.NullPointerException
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2203)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3937)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3941)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4824)
	at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4830)
	at org.grouplens.lenskit.eval.metrics.topn.ItemSelectors$PopularItemSelector.select(ItemSelectors.java:471)
	at org.grouplens.lenskit.eval.traintest.LenskitTestUser$RecommendLoader.load(LenskitTestUser.java:154)
	at org.grouplens.lenskit.eval.traintest.LenskitTestUser$RecommendLoader.load(LenskitTestUser.java:143)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3527)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2319)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2282)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2197)
	... 17 more
Caused by: java.lang.NullPointerException
	at org.grouplens.lenskit.eval.metrics.topn.ItemSelectors$PopularItemSelector.apply(ItemSelectors.java:483)
	at org.grouplens.lenskit.eval.metrics.topn.ItemSelectors$PopularItemSelector.apply(ItemSelectors.java:458)
	at com.google.common.cache.CacheLoader$FunctionToCacheLoader.load(CacheLoader.java:151)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3527)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2319)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2282)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2197)
	... 28 more
```

**Comment by mdekstrand** [comment]
I've prepared a fix in #853 that should fix this problem.

**Comment by mdekstrand** [comment]
The #853 merge is complete. Once the Travis build has finished, you should be able to get the new code by using LensKit version `2.2.2-SNAPSHOT`, from the Maven repository https://oss.sonatype.org/content/repositories/snapshots/.

**Comment by mdekstrand** [comment]
Going to call this fixed in #853.
